### PR TITLE
Accessibility Support for New UI

### PR DESCRIPTION
**Pull request by hghazzi**
### This PR will...

Improve the user experience of navigating the video player and controls via the keyboard, using tab, enter, space and esc

### Why is this Pull Request needed?

To better support accessibility and screen readers

### Are there any Pull Requests open in other repos which need to be merged with this?

https://github.com/jwplayer/jwplayer-plugin-related/pull/163
https://github.com/jwplayer/jwplayer-commercial/pull/4076
https://github.com/jwplayer/jwplayer/pull/2388

#### Addresses Issue(s):

JW8-395

### CODE REVIEW

**Comment by johnBartos** [comment]
@hghazzi Uncommit the lock

File: src/js/view/logo.js [hunk 1]
```diff
@@ -84,7 +84,13 @@ export default function Logo(_model) {
         _img.src = _settings.file;
 
         const logoInteractHandler = new UI(_logo);
-        logoInteractHandler.on('click tap', function (evt) {
+
+        if (_settings.link) {
+            _logo.setAttribute('tabindex', '0');
+            _logo.setAttribute('aria-label', 'Logo');
```
Comment (karimJWP) [inline src/js/view/logo.js]: even if the logo does not have a redirect, would we want the user to know it's there, for branding? Branding is one of the reasons for having a logo on your video. I could go either way on this, @aliciahurst any thoughts ?

File: src/js/view/utils/submenu-factory.js [hunk 2]
```diff
@@ -19,10 +19,15 @@ export const makeSubmenu = (settingsMenu, name, contentItems, icon) => {
     } else {
         const categoryButton = button(`jw-settings-${name}`, () => {
             settingsMenu.activateSubmenu(name);
+            console.log('submenu enter', submenu.element().children[0]);
```
Comment (karimJWP) [inline src/js/view/utils/submenu-factory.js]: @hghazzi are we purposely leaving this console.log?

File: src/js/view/logo.js (hunk 1 above)
Comment (hghazzi) [reply src/js/view/logo.js]: @karimJWP that was part of the acceptance criteria

File: src/js/view/controls/controlbar.js [hunk 3]
```diff
@@ -71,17 +71,17 @@ function createCastButton(castToggle, localization) {
         return;
     }
 
-    const ariaText = localization.cast;
 
     const castButton = document.createElement('button', 'google-cast-button');
-    ariaLabel(castButton, ariaText);
     castButton.setAttribute('type', 'button');
+    castButton.setAttribute('tabindex', '-1');
```
Comment (karimJWP) [inline src/js/view/controls/controlbar.js]: wouldn't this prevent users from tabbing to the cast button ? Why do we want that ?

Comment (hghazzi) [reply src/js/view/controls/controlbar.js]: There's a div around the cast button, so we only want the div to be tabbed to?

File: src/js/view/utils/submenu-factory.js (hunk 2 above)
Comment (hghazzi) [reply src/js/view/utils/submenu-factory.js]: Removed

File: src/js/view/controls/controlbar.js (hunk 3 above)
Comment (karimJWP) [reply src/js/view/controls/controlbar.js]: 👍

**Comment by aliciahurst** [comment]
Tested on PR build 9762, here's everything I found.
* When I tabbed to discovery and hit enter to open it, and hit tab again, I was out of the player.  Expected is when discovery opens the first item is in focus.
* When I hit escape on the discovery overlay it did not close. Expected is the overlay closed.
* I had the whole player focused and the controls faded. Had to pause the video to get them back. Is this how we want it to behave, or when focused, do we want controls to stay up if the entire player is in focus?
* When the menu first opens, I imagine the first submenu is selected with the first item focused as per the requirements.
* "Tabbing advances through the settings menu top nav" I think this may have been unclear. Tabbing should advance the _focus_ through the icons in the submenu. But currently when you hit tab, it changes to that tab, which is a break from the click behavior (it's not on hover, it's on click). You should still have to hit enter to get to that tab, and when you hit enter, the first item in that submenu should be selected. Make sense? 
* In fullscreen, pressing space didn't seem to pause/unpause, and I don't seem to be able to control anything via keyboard at all.
* When I was in the options of one submenu and I kept tabbing through the options, when it reached the end of the text options, instead of going back up to the icons in the menu bar, the focus left the player. So then it was impossible to even close the menu or navigate to other tabs. Expected is it goes back up to the next icon in the submenu.
* When you focus on an icon in the control bar there's no way to tell that you've selected it. So it should have a focus outline.
* When you open the sharing menu or the CC menu the first item should already be in focus.

**Comment by hghazzi** [comment]
@aliciahurst looks like the related changes were not included in this build because they were in the related repo

**Comment by hghazzi** [comment]
test this please

**Pull request merged**